### DISCUSSION TITLE
cannot locate symbol "_ZNSt6__ndk18__rs_getEv"

### DISCUSSION THREAD

**Discussion by MarkHung00** [android/ndk#1782] [Q&A]
We are using Android 11.0 and our ndk is r21 to develop our application, 
Currently we have the following problems
cannot locate symbol "_ZNSt6__ndk18__rs_getEv"
We don't know which libs are missing, or how we can fix it

**Comment by jfgoog** [comment]
There is not enough information here to diagnose this issue. Please provide more information on how to reproduce, er the NDK bug template.

**Comment by stephenhines** [comment]
`std::__ndk1::__rs_get()` is the function that appears to be missing. Are you using NDK RenderScript? I realize that you mentioned using r21, but you should be aware that RenderScript support was removed in r24 (https://github.com/android/ndk/wiki/Changelog-r24#announcements). I suspect that you're not linking the support library for RenderScript, but as @jfgoog said, you need to give us more information.

**Comment by enh-google** [comment]
(despite the name, i think this is actually part of the random_shuffle implementation :-) see `llvm-project/libcxx/src/random_shuffle.cpp`.)

**Reply by pirama-arumuga-nainar** [reply]
... and libc++ should be added automatically by clang and NDK build system.  Are you setting any variable that opt out of libc++/STL  (-nostdlib or such)?
